lib/extensions/iiif_manifest/manifest_builder/deep_file_set_enumerator/nested_each.rb

```diff
@@ -1,18 +1,25 @@
-# unmodified from iiif_manifest
+# modified from iiif_manifest to support mixed file_sets, works in sequence
 module Extensions
   module IIIFManifest
     module ManifestBuilder
       module DeepFileSetEnumerator
         module NestedEach
-          # lists all direct file_set members, then recurs through work members
+          # lists all direct file_set members and member works' file_sets, interspersed
           def each(&block)
-            file_set_presenters.each do |file_set_presenter|
-              yield file_set_presenter
-            end
-            work_presenters.each do |work_presenter|
-              self.class.new(work_presenter).each(&block)
+            member_presenters.each do |member_presenter|
+              if member_presenter.id.in? file_set_presenters.map(&:id)
+                yield file_set_presenters.find { |fp| fp.id == member_presenter.id }
+              elsif member_presenter.id.in? work_presenters.map(&:id)
+                self.class.new(work_presenters.find { |wp| wp.id == member_presenter.id } ).each(&block)
```

Comment:
Are the multiple calls to `file_set_presenters` or `work_presenters` going to cause repeated builds of the presenter objects or their solr docs?

Reply:
It shouldn't -- the method ultimately called should only build them the first time, via the nil guard:
https://github.com/samvera/iiif_manifest/blob/v0.5.0/lib/iiif_manifest/manifest_builder/deep_canvas_builder_factory.rb#L29-L31
https://github.com/samvera/hyrax/blob/v2.9.6/app/presenters/hyrax/work_show_presenter.rb#L176
https://github.com/samvera/hyrax/blob/v2.9.6/app/presenters/hyrax/member_presenter_factory.rb#L29-L37

I did have another version of this loop that was more concisely written and should have been logically equivalent, but it seemed to misbehave spookily, so I reverted to this less elegant version.

Reply:
It's meant to work that way, but I think [this line](https://github.com/samvera/hyrax/blob/v2.9.6/app/presenters/hyrax/work_show_presenter.rb#L280-L282) that creates a new factory object makes the memoization in the factory useless.

Reply:
Oh, snap!  What about patching that line to the following?
```ruby
      def member_presenter_factory
        @member_presenter_factory ||= MemberPresenterFactory.new(solr_document, current_ability, request)
      end
 ```

Reply:
Seems reasonable